Add accept/cancel leader bindings to Forge and Magit log modes

yuhan0 opened the pull request:
This allows one to use `,c` and `,k` in place of `C-c C-c` and `C-c C-k` in the Magit log mode (e.g. when choosing a commit to fixup into) , and in the Forge post buffer for pull requests and issues.

Miciah commented:
Oops, I was about to make a similar PR when I saw this! https://github.com/Miciah/spacemacs/commit/aec2082d94062f46132ce58571536fe0c441872f

The documentation needs to be updated; feel free to copy from the linked commit.

My commit also adds bindings for forge-topic mode; if you like, feel free to copy those as well.

yuhan0 commented:
Thanks @Miciah, I added your changes and edited a bit in the Readme for commit messages editing which had a mistakenly added table. I also removed references to the default `C-c C-c` / `C-c C-k` bindings, wonder if it's necessary to mention them as the documentation as a whole is very vim-style oriented.

yuhan0 commented:
Also found out that `evil-magit` binds `gj` and `gk` to cycle through commit messages, added that to the readme as well.

Miciah commented on layers/+source-control/git/README.org:
```diff
@@ -194,16 +195,25 @@ intended. To work around this, you can use =v= instead (since Magit only stages
+| ~SPC m c~ or ~SPC m ,~ | commit changes with the entered message                   |
+| ~SPC m a~ or ~SPC m k~ | discard message and abort the commit                      |
+| ~g j~                  | cycle through history to the previous commit message      |
+| ~g k~                  | save current commit message and cycle to the next message |
```
In the commit edit buffer, I see that <kbd>g j</kbd> is bound to `evil-next-visual-line`, and <kbd>g k</kbd> is bound to `evil-previous-visual-line`.  Are they bound differently for you? I do see that <kbd>M-n</kbd> is `git-commit-next-message`, and <kbd>M-p</kbd> is `git-commit-prev-message`.

Miciah replied:
I see that evil-magit binds <kbd>&lt;visual state&gt; g j</kbd> to `git-commit-next-message`, but <kbd>g j</kbd> is `evil-next-visual-line` and takes precedence for me in both the vim editing style and the hybrid editing style.

yuhan0 replied:
That's interesting - I'm on vim editing style and `g j` / `g k` are bound to the commit cycling functions, whereas `M-n` and `M-p` are unbound.

yuhan0 replied:
Can anyone else using Evil + Magit confirm if the `gj/gk` or `M-n/M-p` bindings work for them in the commit message editing buffer? 

(This isn't part of the original PR's scope but I thought I'd just lump it in, since they take so long to get merged around here)

sdwolfz replied:
@yuhan0 I see the same thing as @Miciah in my setup.

yuhan0 commented:
I tried looking more into that, it turned out that ~M-n~ and ~M-p~ were being overriden somehow by the evil-mc package, but I couldn't reproduce the `gj` and `gk` being shadowed by `evil-visual-line`, even from a minimal installation with only the `git` layer enabled.

Anyway, I simply included both bindings in the documentation (it's a really useful functionality that I didn't know existed before this)

duianto commented:
Thank you for contributing to Spacemacs!
The PR has been cherry-picked into the `develop` branch, you can safely delete your branch.
